libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +58,43 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	firewalldRunning, err = checkRunning()
+
+	if err != nil {
+		return err
+	}
+	// If firewalld is not running on the first check poll
 	if !firewalldRunning {
-		connection.sysconn.Close()
-		connection = nil
+		tick := time.NewTicker(500 * time.Millisecond)
+		timeLapse := time.Duration(0)
+		defer tick.Stop()
+
+	TickLoop:
+		for {
+			select {
+			case <-tick.C:
+				firewalldRunning, _ = checkRunning()
+				timeLapse += 500 * time.Millisecond
+
+				log.G(context.TODO()).Debugf("Checking status of Firewalld")
+
+				if firewalldRunning || timeLapse >= FireWalldInitializationTimeout {
+					break TickLoop
+				}
+
+			default:
+				continue TickLoop
+			}
+		}
+	}
+
+	if !firewalldRunning && err == nil {
```

Comment:
I don't think `err` can ever be non-nil here. 

Also; always be cautions when combining conditions; these can be a real footgun. In this case; we'd log "Firewalld is currently in state running" if

- `firewalldRunning == false` and `err != nil`
- `firewalldRunning == true` and `err != nil`

(It's why golang wants you to handle errors; if a function (`checkRunning()`) returns an error, then we must assume any value returned cannot be used (with some rare exceptions).

Reply:
I believe that is the point, in both of those cases firewalld would be in an INIT state.  The exact acceptance criteria I have says:

"When we hit the limit, **we should act as firewalld is ready (e.g. that it left init)** we log a message and continue, and hope for the best"